src/main/java/core/basesyntax/dao/WriteToFile.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.products.Fruit;
+import core.basesyntax.storage.ListStorage;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class WriteToFile {
+    public boolean writeToFile(String filePath) {
+        Path path = Paths.get(filePath);
+        if (Files.exists(path)) {
+            try {
+                Files.delete(path);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            createFile(filePath);
+        } else {
+            createFile(filePath);
+        }
+
+        Map<String, Integer> mapToFile = ListStorage.listStorage.stream()
+                .collect(Collectors.groupingBy(Fruit::getName,
+                        Collectors.summingInt(Fruit::getAmount)));
+
+        List<String> stringsToFile = new ArrayList<>();
+        stringsToFile.add("fruit,quantity");
+
+        for (Map.Entry<String, Integer> entry : mapToFile.entrySet()) {
+            stringsToFile.add(entry.getKey() + "," + entry.getValue());
+        }
+        try {
+            Files.write(path, stringsToFile, StandardOpenOption.APPEND);
+        } catch (IOException e) {
+            throw new RuntimeException("Error with writing to file. "
+                    + "Rerun main. ", e);
+        }
+        return true;
+    }
+
+    private void createFile(String filePath) {
+        Path path = Paths.get(filePath);
+        try {
+            Files.createFile(path);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
We don't need this method. File will be created automatically when calling Files.write()

Reply:
It has to be like this, because I do Files.write(path, stringsToFile, StandardOpenOption.APPEND); it will not erase the content of file but will append to old data. I write to file directly from List without any conversion. Plus it is better to create new file for fresh data, it will be more informative to see when this file was created :)

Reply:
Sorry, but I can't agry with you about appending new data to old data. You are deleting old file (if it exist) at line 16.
Also you can use `Files.deleteIfExists(path);`